test/test_qn_model.jl

```diff
@@ -0,0 +1,59 @@
+using LinearOperators
+
+function check_qn_model(qnmodel)
+
+  model = qnmodel.model
+  @assert typeof(qnmodel) <: NLPModels.QuasiNewtonModel
+  @assert qnmodel.meta.nvar == model.meta.nvar
+  @assert qnmodel.meta.ncon == model.meta.ncon
+
+  x = rand(qnmodel.meta.nvar)
+
+  @assert obj(model, x) == obj(qnmodel, x)
+  @assert neval_obj(model) == 2
+
+  @assert all(grad(model, x) == grad(qnmodel, x))
+  @assert neval_grad(model) == 2
+
+  @assert all(cons(model, x) == cons(qnmodel, x))
+  @assert neval_cons(model) == 2
+
+  @assert all(jac(model, x) == jac(qnmodel, x))
+  @assert neval_jac(model) == 2
+
+  v = rand(qnmodel.meta.nvar)
+  u = rand(qnmodel.meta.ncon)
+
+  @assert all(jprod(model, x, v) == jprod(qnmodel, x, v))
+  @assert neval_jprod(model) == 2
+
+  @assert all(jtprod(model, x, u) == jtprod(qnmodel, x, u))
+  @assert neval_jtprod(model) == 2
+
+  H = hess_op(model, x)
+  @assert typeof(H) <: LinearOperators.AbstractLinearOperator
+  @assert size(H) == (model.meta.nvar, model.meta.nvar)
+  @assert all(H * v == hprod(model, x, v))
```

Comment:
This should also be `qnmodel` and `==` verifies all components. `all` is not necessary.